### PR TITLE
Codejam virtual keyboard

### PR DESCRIPTION
**Pull request by OAliona**
1. task - https://github.com/rolling-scopes-school/tasks/blob/master/tasks/codejam-virtual-keyboard.md
2. screenshot - 
codejam-virtual-keyboard

3. deadline: 2019-11-06
4. Total
- generation of DOM elements and index.html is empty by default: +20
- clicking button on physical keyboard makes virtual keyboard button highlighted: +10
- clicking button on virtual keyboard leads to appearance of that button at the input text field: +15
- realization of animation for button click: +15
- usage of eslint: +10
- requirements to repo, commits, and PR are observed: +10

mark calculation:
20+10+15+15+10+10=80

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    } else if (button === 'Tab') {
+      textfield.value += '    ';
+    } else if (button === '&uarr;') {
+      textfield.value += '▲';
+    } else if (button === '&uarr;') {
+      textfield.value += '▲';
+    } else if (button === '&larr;') {
+      textfield.value += '◄';
+    } else if (button === '&rarr;') {
+      textfield.value += '►';
```
Comment (YauheniX) [inline src/index.js]: you can use map for this

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    return function hanleButtonClick() {
+      if (isSpecialChar(button)) {
+        handleSpecialChar(button);
+      } else {
+        textfield.value += button;
+      }
+    };
+  }
+
+  for (let rowIndex = 0; rowIndex < buttons.length; rowIndex += 1) {
```
Comment (YauheniX) [inline src/index.js]: foreach

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    const row = buttons[rowIndex];
```
Comment (YauheniX) [inline src/index.js]: foreach

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    const rowElement = document.createElement('div');
+    rowElement.classList.add('line');
+
+    for (let buttonIndex = 0; buttonIndex < row.length; buttonIndex += 1) {
```
Comment (YauheniX) [inline src/index.js]: foreach

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      buttonElement.addEventListener('click', buttonClick(row[buttonIndex]));
+      rowElement.appendChild(buttonElement);
+    }
+
+    document.getElementsByClassName('keyboard')[0].appendChild(rowElement);
+  }
+}
+
+function mapKey(key) {
+  switch (key) {
```
Comment (YauheniX) [inline src/index.js]: the same

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    buttons[0].classList.add('button-active');
+    buttons[0].dispatchEvent(new Event('click'));
+  }
+}
+
+function handleKeyUp(event) {
+  const key = mapKey(event.key);
+  const buttons = document.getElementsByClassName(`button-${key}`);
+
+  if (buttons.length > 0) {
```
Comment (YauheniX) [inline src/index.js]: if (buttons.length)

Comment (OAliona) [reply src/index.js]: Done

File: src/index.js (hunk 5 above)
Comment (OAliona) [reply src/index.js]: Done

File: src/index.js (hunk 4 above)
Comment (OAliona) [reply src/index.js]: Done

File: src/index.js (hunk 3 above)
Comment (OAliona) [reply src/index.js]: Done

File: src/index.js (hunk 2 above)
Comment (OAliona) [reply src/index.js]: Done

File: src/index.js (hunk 1 above)
Comment (OAliona) [reply src/index.js]: Done

**Comment by YauheniX** [comment]
Final score:80

**Pull request closed without merging**